src/observers/link_prefetch_observer.js

```diff
@@ -77,12 +90,23 @@ export class LinkPrefetchObserver {
         )
 
         prefetchCache.setLater(location.toString(), fetchRequest, this.#cacheTtl)
-
-        link.addEventListener("mouseleave", () => prefetchCache.clear(), { once: true })
       }
     }
   }
 
+  #tryToCancelPrefetchRequest = (event) => {
+    const targets = "targetTouches" in event ?
+      [event.target, ...event.targetTouches] :
+      [event.target]
+    const leavingPrefetchedLink = targets.some((target) => target === this.#prefetchedLink)
```

Comment:
Is this correct for touch events? If the `touchend` event still has a `targetTouch` that is the prefetched link, that means we didn't leave the link, right? Same with `touchmove`?

Reply:
Maybe [TouchEvent.changedTouches](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/changedTouches) is a better candidate:

> * For the [touchstart](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchstart_event) event, it is a list of the touch points that  became active with the current event.
> * For the [touchmove](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchmove_event) event, it is a list of the touch points that  have changed since the last event.
> * For the [touchend](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchend_event) event, it is a list of the touch points that have been removed from the surface (that is, the set of touch points corresponding to fingers no longer touching the surface).

Reply:
Not sure if that will work. For example, if a `touchend` fires and `changedTouches` points to another element, that doesn't mean the link is not still being tapped. It could be another finger that was removed from the surface. Same with `touchmove`.

Maybe I'm missing something, but I think we'll have to use two different handlers. For example `#checkIfPrefetchValidAfterMouseLeave` and `#checkIfPrefetchValidAfterTouchChange`. The two cases have opposite logic: for `mouseleave` we left the link if the event target is the link; for touch changes we left the link if _none_ of the targets is the link.

Reply:
After more reflection, I realize that I've misread the conditionals in https://github.com/hotwired/turbo/commit/5d15bb3529b63efb30a7f1eb90c17be617820445.

I've pushed up a new commit use `Array.some` to check a `touchend` event's `changedTouches` for **the presence** of the link, `Array.some` to check for a `touchmove` event's `targetTouches` for **the absence** of the link, and all other types of events for the target itself.

Even though its a singular conditional and a singular event listener callback, it isn't as general purpose as I originally intended.

With that being acknowledged, I think that if a change accounts for touch events, removes all event listeners that it attaches, and doesn't leak additional listeners, it's worth shipping. I don't have a preference between this PR and https://github.com/hotwired/turbo/commit/5d15bb3529b63efb30a7f1eb90c17be617820445. If it's easier for you to coordinate the necessary steps to merge your fix, I say go for it.